test/e2e/network_test.go

```diff
@@ -38,109 +37,108 @@ var (
-		if err == nil {
-			break
-		}
-	}
-	require.NoError(t, err, "could not create Windows Server deployment on first Node")
-
-	// Get the pod so we can use its IP
-	winServerIP, err := testCtx.getPodIP(*winServerDeploymentOnFirstNode.Spec.Selector)
-	require.NoError(t, err, "could not retrieve pod with selector %v", *winServerDeploymentOnFirstNode.Spec.Selector)
-
-	// test Windows <-> Windows across nodes
-	winCurlerJobOnSecondNode, err := testCtx.createWinCurlerJob(strings.ToLower(secondNode.Status.NodeInfo.MachineID), winServerIP)
-	require.NoError(t, err, "could not create Windows job on second Node")
-
-	// This is prone to timing out due to having to pull the Windows image so trying multiple times
-	for i := 0; i < 10; i++ {
-		err = testCtx.waitUntilJobSucceeds(winCurlerJobOnSecondNode.Name)
-		if err == nil {
-			break
-		}
-	}
-	assert.NoError(t, err, "could not curl the Windows webserver pod on the first node from Windows container "+
-		"on the second node")
-
-	// delete the deployment and job created
-	if err = testCtx.deleteDeployment(winServerDeploymentOnFirstNode.Name); err != nil {
-		t.Logf("could not delete deployment %s", winServerDeploymentOnFirstNode.Name)
+	testCases := []struct {
+		name            string
+		curlerOS        osTypes
+		useClusterIPSVC bool
+	}{
+		{
+			name:            "linux and windows",
+			curlerOS:        linux,
+			useClusterIPSVC: false,
+		},
+		{
+			name:            "windows and windows",
+			curlerOS:        windows,
+			useClusterIPSVC: false,
+		},
+		{
+			name:            "linux and windows through a clusterIP svc",
+			curlerOS:        linux,
+			useClusterIPSVC: true,
+		},
+		{
+			name:            "windows and windows through a clusterIP svc",
+			curlerOS:        windows,
+			useClusterIPSVC: true,
+		},
 	}
+	require.Greater(t, len(gc.nodes), 0, "test requires at least one Windows node to run")
+	firstNodeAffinity, err := getAffinityForNode(&gc.nodes[0])
+	require.NoError(t, err, "could not get affinity for node")
 
-	if err = testCtx.deleteJob(winCurlerJobOnSecondNode.Name); err != nil {
-		t.Logf("could not delete job %s", winCurlerJobOnSecondNode.Name)
+	for _, node := range gc.nodes {
+		t.Run(node.Name, func(t *testing.T) {
+			affinity, err := getAffinityForNode(&node)
+			require.NoError(t, err, "could not get affinity for node")
+
+			// Deploy a webserver pod on the new node. This is prone to timing out due to having to pull the Windows image
+			// So trying multiple times
+			var winServerDeployment *appsv1.Deployment
+			for i := 0; i < deploymentRetries; i++ {
+				winServerDeployment, err = testCtx.deployWindowsWebServer("win-webserver-"+strings.ToLower(node.Status.NodeInfo.MachineID), affinity)
+				if err == nil {
+					break
+				}
+			}
+			require.NoError(t, err, "could not create Windows Server deployment")
+			defer testCtx.deleteDeployment(winServerDeployment.Name)
```

Comment:
Is there a reason to move from deletion at the [end](https://github.com/openshift/windows-machine-config-operator/pull/129/files#diff-96873a210ce97729b66ec711eb21c852L79) to `defer` ?

Reply:
Yes, this way we still clean up resources even when we fail the test